RadixWallet/Features/DappInteractionFeature/Interactor/DappInteractor.swift

```diff
@@ -311,34 +308,35 @@ struct DappInteractor: Sendable, FeatureReducer {
 	) -> Effect<Action> {
 		.run { send in
 
-			// In case of transaction response, sending it to the peer client is a silent operation.
+			// In case of transaction/preAuthorization responses, sending it to the peer client is a silent operation.
 			// The success or failures is determined based on the transaction polling status.
-			let txID: TransactionIntentHash? = {
-				if case let .success(successResponse) = responseToDapp,
-				   case let .transaction(txID) = successResponse.items
-				{
-					return txID.send.transactionIntentHash
+			let isPersonaDataResponse = {
+				guard case let .success(successResponse) = responseToDapp else {
+					return true
+				}
+				switch successResponse.items {
+				case .authorizedRequest, .unauthorizedRequest:
+					return true
+				case .transaction, .preAuthorization:
+					return false
 				}
-				return nil
 			}()
-			let isTransactionResponse = txID != nil
 
 			do {
 				_ = try await dappInteractionClient.completeInteraction(.response(.dapp(responseToDapp), origin: request.route))
-				if !isTransactionResponse {
+				if isPersonaDataResponse {
 					await send(.internal(
-						.sentResponseToDapp(
+						.sentPersonaDataResponseToDapp(
 							responseToDapp,
 							for: request,
-							dappMetadata,
-							txID
+							dappMetadata
 						)
 					))
 				} else {
-					loggerGlobal.notice("Not delegating to `sentResponseToDapp`")
+					loggerGlobal.notice("Not delegating to `sentPersonaDataResponseToDapp`")
 				}
 			} catch {
-				if !isTransactionResponse {
+				if isPersonaDataResponse {
```

Comment:
I think if we fail to send back the SignedSubintent we should show the `failedToSendResponseToDapp`. We don't show the error for the transaction because it will be submitted to the network by the wallet itself, but in case of the PreAuth the transaction will be submitted by the dApp.
So, if we do fail to send back the response, instead of waiting for it to expire we should show the error eagerly. Now, this will have bigger implications, potentially requiring to strip out the polling view from the TransactionReview, not entirely sure if we should do it, but worth investigating the full implications.